code/game/machinery/autolathe.dm

```diff
@@ -153,84 +136,81 @@
-			return
-
 		being_built = design
+		var/is_stack = ispath(being_built.build_path, /obj/item/stack)
+		var/coeff = (is_stack ? 1 : creation_efficiency) // Stacks are unaffected by production coefficient
 
 		var/multiplier = round(text2num(params["multiplier"]))
-
 		if(!multiplier || !IS_FINITE(multiplier))
 			return
-
-		var/is_stack = ispath(being_built.build_path, /obj/item/stack)
 		multiplier = clamp(multiplier, 1, 50)
 
-		var/coeff = (is_stack ? 1 : creation_efficiency) // Stacks are unaffected by production coefficient
-		var/total_amount = 0
-
-		for(var/material in being_built.materials)
-			total_amount += being_built.materials[material]
-
-		var/power = max(active_power_usage, (total_amount)*multiplier/5) // Change this to use all materials
-
-		var/datum/component/material_container/materials = GetComponent(/datum/component/material_container)
-
+		//check for materials
 		var/list/materials_used = list()
 		var/list/custom_materials = list() // These will apply their material effect, should usually only be one.
-
 		for(var/mat in being_built.materials)
 			var/datum/material/used_material = mat
-			var/amount_needed = being_built.materials[mat] * coeff * multiplier
 
+			var/amount_needed = being_built.materials[mat]
 			if(istext(used_material)) // This means its a category
 				var/list/list_to_show = list()
-
+				//list all materials in said category
 				for(var/i in SSmaterials.materials_by_category[used_material])
 					if(materials.materials[i] > 0)
 						list_to_show += i
-
+				//ask user to pick specific material from list
 				used_material = tgui_input_list(
 					usr,
 					"Choose [used_material]",
 					"Custom Material",
 					sort_list(list_to_show, GLOBAL_PROC_REF(cmp_typepaths_asc))
 				)
-
 				if(isnull(used_material))
-					// Didn't pick any material, so you can't build shit either.
 					return
-
+				//the item composition will be made of these materials
 				custom_materials[used_material] += amount_needed
-
 			materials_used[used_material] = amount_needed
 
-		if(materials.has_materials(materials_used))
-			busy = TRUE
-			to_chat(usr, span_notice("You print [multiplier] item(s) from the [src]"))
-			use_power(power)
-			icon_state = "autolathe_n"
-			var/time = is_stack ? 32 : (32 * coeff * multiplier) ** 0.8
-			addtimer(CALLBACK(src, PROC_REF(make_item), power, materials_used, custom_materials, multiplier, coeff, is_stack, usr), time)
-			. = TRUE
-		else
-			to_chat(usr, span_alert("Not enough materials for this operation."))
+		if(!materials.has_materials(materials_used, coeff, multiplier))
+			say("Not enough materials for this operation!.")
+			return
+
+		//use power
+		var/total_amount = 0
+		for(var/material in being_built.materials)
+			total_amount += being_built.materials[material]
+		var/power = max(active_power_usage, (total_amount) * multiplier / 5) // Change this to use all materials
```

Comment:
Does this still need to be changed?

Reply:
That comment is outdated so no nothing needs to be changed, i copied & pasted it by accident I'll remove it